MacroPlugin/Decl/HalleyCodingKeysGenerator.swift

```diff
@@ -0,0 +1,33 @@
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+struct HalleyCodingKeysGenerator {
+
+    let properties: [HalleyPropertyDecl]
+    let isPublic: Bool
+
+    func generate() -> EnumDeclSyntax {
+        return EnumDeclSyntax(
+            modifiers: isPublic ? DeclModifierListSyntax { DeclModifierSyntax(name: .keyword(.public)) } : [],
+            name: "CodingKeys",
+            inheritanceClause: InheritanceClauseSyntax {
+                InheritedTypeSyntax(type: TypeSyntax(stringLiteral: "String"))
+                InheritedTypeSyntax(type: TypeSyntax(stringLiteral: "CodingKey"))
+                InheritedTypeSyntax(type: TypeSyntax(stringLiteral: "IncludeKey"))
+            }
+        ) {
+            // Some properties can be skipped via @HalleySkip attribute
```

Comment:
I think this comment is outdated or something? Skipping is done by providing `nil` argument? 

But I actually like `HalleySkip`